### PR TITLE
[RDF] Refactor RDisplay::AsString and Print

### PR DESCRIPTION
**Pull request by silverweed**
# This Pull request:
Refactors `RDisplay::Print` to use `RDisplay::AsString` internally rather than duplicating logic. Removes some small differences from Print's and AsString's output but preserves the "dotting" logic for eliding lines and columns in Print().

## Checklist:

- [x] tested changes locally
- [ ] updated the docs (if necessary)

### CODE REVIEW

File: tree/dataframe/src/RDFDisplay.cxx
Comment (silverweed) [reply tree/dataframe/src/RDFDisplay.cxx:252]: In principle we might want to make `considerDots` an option orthogonal to Print vs AsString. Delegating the decision whether to print the messages or not to the caller allows to change this more easily, and since this code doesn't have strict performance requirements I think we can accept some wasted work when we (currently) ignore the messages in AsString.

Comment (silverweed) [reply tree/dataframe/src/RDFDisplay.cxx:252]: >  It is also an order of printing change.

You mean that all warnings will now be grouped after the regular printing? That's true, I didn't consider that and I suppose we want to preserve the current printing behavior...I'll adopt your solution, thanks.

**Review by vepadulano — APPROVED** [review]
Thank you! I believe the changes look good and this is a nice improvement!

**Comment by vepadulano** [comment]
Maybe consider applying the changes suggested by clang-format before merging

**Review by martamaja10 — APPROVED** [review]
Thanks Jack!

File: tree/dataframe/src/RDFDisplay.cxx
Comment (pcanal) [reply tree/dataframe/src/RDFDisplay.cxx]: @silverweed @vepadulano Was this last comment discussed offline?

Comment (silverweed) [reply tree/dataframe/src/RDFDisplay.cxx]: Sorry Philippe, I didn't see your last comment so I thought everything was good. But you're clearly right, the order is not preserved even with your initial solution. The `ToStream` solution sounds good, I will open a PR with that change. Apologies for not addressing this here.

Comment (silverweed) [reply tree/dataframe/src/RDFDisplay.cxx]: Although even with the `ToStream` I would have to either pass 2 streams or filter out the warnings from `AsString()`...so I think the real solution will be slightly more complex.

Comment (silverweed) [reply tree/dataframe/src/RDFDisplay.cxx]: #17173

Comment (pcanal) [reply tree/dataframe/src/RDFDisplay.cxx]: > as the warnings are already all printed before the output 

I am not quite sure how beside maybe the difference in flushing between stderr and stdout, but that's not important.  The important part was to make a decision on the intended behavior and document it.

> Sorry Philippe, I didn't see your last comment

I make it an habit to not merge a PR until all discussion are marked as resolved to avoid missing some of the comments.  

The [Celeritas](https://github.com/celeritas-project/celeritas) team wrote a reasonable formal-ish description on how to handle those resolution.
```
When you've fully implemented the reviewer's comment, you may mark it as resolved without commenting. Do not resolve a conversation if you disagree with the feed: instead, post your view in a follow-on comment and wait for the reviewer to respond. If you comment, whether to supplement your change or to iterate with the reviewer, please do not resolve the conversation since that makes it hard to find your comment.
```

Comment (silverweed) [reply tree/dataframe/src/RDFDisplay.cxx]: > I make it an habit to not merge a PR until all discussion are marked as resolved to avoid missing some of the comments.

I wish Github had a good way to handle reviews, I still haven't found any system that works better than Helix Swarm on this. One can dream.

> I am not quite sure how beside maybe the difference in flushing between stderr and stdout

Simply, all the `Info` calls come before any `cout` call in the original code, they are fully disjoint. And they should stay that way because otherwise the table output would be messed up by intermingled warnings, which is undesirable.

Comment (pcanal) [reply tree/dataframe/src/RDFDisplay.cxx]: > Simply, all the Info calls come before any cout call in the original code, 

That was not obvious from the diff :)

> And they should stay that way because otherwise the table output would be messed up by intermingled warnings, which is undesirable.

This makes sense.  Thanks.

**Pull request merged**
